apps/site/pages/en/learn/modules/package-configuration.md

```diff
@@ -0,0 +1,563 @@
+  "type": "commonjs",
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": {
+      "node": "./dist/cjs/index.js",
+      "default": "./dist/esm/index.mjs"
+    },
+    "./package.json": "./package.json"
+  }
+}
+```
+
+### ESM source and distribution
+
+The wine of packages: Simple, tried, and true.
+
+Note that since Node.js v23.0.0, it is possible to `require` static ESM (code that does not use top-level `await`). See [Loading ECMAScript modules using `require()`](https://nodejs.org/api/modules.html#loading-ecmascript-modules-using-require) for details.
+
+This is almost exactly the same as the CJS-CJS configuration above with 1 small difference: the [`"type"`](https://nodejs.org/api/packages.html#type) field.
+
+**Working example**: [esm-with-esm-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/esm/esm-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "esm-source-and-distribution",
+  "type": "module"
+  // "main": "./index.js"
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "esm-source-and-distribution",
+  "type": "module",
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": "./dist/index.js",
+    "./package.json": "./package.json"
+  }
+}
+```
+
+Note that ESM now _is_ “backwards” compatible with CJS: a CJS module now _can_ [`require()` an ES Module](https://nodejs.org/api/modules.html#loading-ecmascript-modules-using-require) without a flag as of 23.0.0 and 22.12.0.
+
+### ESM source with only CJS distribution
+
+We're not in Kansas anymore, Toto.
+
+The configurations (there are 2 options) are nearly the same as [ESM source and both CJS & ESM distribution](#esm-source-and-both-cjs-amp-esm-distribution), just exclude `packageJson.exports.import`.
+
+💡 Using `"type": "module"`[^2] paired with the `.cjs` file extension (for commonjs files) yields best results. For more information on why, see [Down the rabbit-hole](#down-the-rabbithole) and [Gotchas](#gotchas) below.
+
+**Working example**: [esm-with-cjs-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/esm/cjs-distro)
+
+### ESM source and both CJS & ESM distribution
+
+These are "mixologist" territory.
+
+When source code is written in non-JavaScript (ex TypeScript), options can be limited due to needing to use file extension(s) specific to that language (ex `.ts`) and there may be no `.mjs` equivalent.
+
+Similar to [CJS source and both CJS & ESM distribution](#cjs-source-and-both-cjs-amp-esm-distribution), you have the same options.
+
+#### Publish only a CJS distribution with property exports
+
+The "Mojito" of packages: Tricky to make and needs good ingredients.
+
+This option is almost identical to the [CJS source with CJS & ESM distribution's property exports](#attach-named-exports-directly-onto-raw-exports-endraw-) above. The only difference is in package.json: `"type": "module"`.
```

Comment:
I meant that this only shows `"type": "module"` and `.cjs` files in the package.json. It should explain that somewhere in this package's source form, there are `.js` files containing ESM. Otherwise one wonders why is  `"type": "module"`  needed at all (you could also have all your source files in `.mjs` and then ship `"type": "commonjs"` + ".js" containing CJS)